### PR TITLE
Sort integers wrt set bits

### LINKED ISSUE #3960: Sort integers w.r.t the number of 1 bits in ascending order | Bit Manipulation | GSSoC'21 | CPP/C++
## 💥 Proposal

To implement the program Sort integers w.r.t the number of 1 bits in ascending order in C++/ CPP.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?
YES

### PR DESCRIPTION
**Pull request by Harikrishnan6336**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

I have added the source code of sort integers w.r.t set bits in C++ as part of GSSoC'21

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #3960

### CODE REVIEW

File: C-Plus-Plus/README.md [hunk 1]
```diff
@@ -319,6 +319,7 @@ _add list here_
 - [Two odd occuring](Bit--Manipulation/Two_odd_occuring.cpp)
 - [Power_Set_Using_Bitwise](Bit--Manipulation/Power_Set_Using_Bitwise.cpp)
 - [Reverse Bits](Bit--Manipulation/reverse_bits.cpp)
+- [Sort integers w.r.t thier number of set bits](Bit--Manipulation/sort_integers_w.r.t_no_of_bits.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Use a better filename

File: C-Plus-Plus/Bit--Manipulation/sort_integers_w.r.t_no_of_bits.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
+// C++ program to sort integers w.r.t the number of set bits in its binary equivalence
```
Comment (HarshCasper) [inline C-Plus-Plus/Bit--Manipulation/sort_integers_w.r.t_no_of_bits.cpp]: Can you add more description?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by Harikrishnan6336** [comment]
@HarshCasper I have made the required changes. Kindly review my PR.

File: C-Plus-Plus/README.md [hunk 3]
```diff
@@ -319,6 +319,7 @@ _add list here_
 - [Two odd occuring](Bit--Manipulation/Two_odd_occuring.cpp)
 - [Power_Set_Using_Bitwise](Bit--Manipulation/Power_Set_Using_Bitwise.cpp)
 - [Reverse Bits](Bit--Manipulation/reverse_bits.cpp)
+- [Sort integers w.r.t thier number of set bits](Bit--Manipulation/sort_integers_by_thier_num_of_set_bits.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Typos. Use full words instead of short form

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/README.md [hunk 4]
```diff
@@ -319,6 +319,7 @@ _add list here_
 - [Two odd occuring](Bit--Manipulation/Two_odd_occuring.cpp)
 - [Power_Set_Using_Bitwise](Bit--Manipulation/Power_Set_Using_Bitwise.cpp)
 - [Reverse Bits](Bit--Manipulation/reverse_bits.cpp)
+- [Sort integers by the number of set bits present in them](Bit--Manipulation/sort_integers_by_thier_number_of_set_bits.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: `thier` is a typo error. It should be `their`

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**